### PR TITLE
[visionOS] Replace `TARGET_OS_XR` with `TARGET_OS_VISION`

### PR DESCRIPTION
**Pull request by andrewheard**
In Xcode 15 beta 5, `TARGET_OS_XR` was replaced with `TARGET_OS_VISION` in ObjC and `os(xrOS)` was replaced with `os(visionOS)` in Swift.

### CODE REVIEW

**Comment by andrewheard** [comment]
This builds for the Vision Pro Simulator in Xcode 15 beta 5 when setting the `GoogleDataTransport` and `GoogleUtilities` [dependencies in `Package.swift`](https://github.com/firebase/firebase-ios-sdk/blob/73c646a6be9b196e90b42ea2436052a5012603de/Package.swift#L149-L156) to `branch: "nc/xcode15b5"`, which contain the same `TARGET_OS_XR` to `TARGET_OS_VISION` refactor for those repos.

File: FirebaseCore/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Firebase 10.13.0
+- For developers building for visionOS, Xcode beta 5 or later is required.
```
Comment (paulb777) [inline FirebaseCore/CHANGELOG.md:2]: Xcode 15 beta 5

File: FirebaseCore/Tests/Unit/FIRFirebaseUserAgentTests.m [hunk 2]
```diff
@@ -35,12 +35,21 @@ - (void)testFirebaseUserAgent_ApplePlatformFlag {
   XCTAssertFalse([[FIRApp firebaseUserAgent] containsString:@"apple-platform/macos"]);
   XCTAssertFalse([[FIRApp firebaseUserAgent] containsString:@"apple-platform/watchos"]);
   XCTAssertTrue([[FIRApp firebaseUserAgent] containsString:@"apple-platform/maccatalyst"]);
+  XCTAssertFalse([[FIRApp firebaseUserAgent] containsString:@"apple-platform/visionos"]);
```
Comment (paulb777) [inline FirebaseCore/Tests/Unit/FIRFirebaseUserAgentTests.m:38]: Nice add!

**Review by paulb777 — APPROVED** [review]

**Comment by paulb777** [comment]
Crashlytics/catalyst is a common flake

File: FirebaseCore/CHANGELOG.md (hunk 1 above)
Comment (andrewheard) [reply FirebaseCore/CHANGELOG.md]: That's an important distinction, thanks for catching!

**Pull request merged**